File: data/derived_cot/rq1_traces/translation/glm5.1/py_to_cpp/code_output/CookiesUtil.cpp
#include <string>
#include <fstream>
#include <nlohmann/json.hpp>

using json = nlohmann::json;

class CookiesUtil {
public:
    std::string cookies_file;
    json cookies;

    CookiesUtil(const std::string& cookies_file)
        : cookies_file(cookies_file), cookies(nullptr) {}

    void get_cookies(const json& reponse) {
        cookies = reponse.at("cookies");
        _save_cookies();
    }

    json load_cookies() {
        std::ifstream file(cookies_file);
        if (!file.is_open()) {
            return json::object();
        }
        return json::parse(file);
    }

    bool _save_cookies() {
        try {
            std::ofstream file(cookies_file);
            if (!file.is_open()) {
                return false;
            }
            file << cookies.dump();
            return true;
        } catch (...) {
            return false;
        }
    }

    void set_cookies(json& request) {
        std::string cookie_str;
        for (auto it = cookies.begin(); it != cookies.end(); ++it) {
            if (!cookie_str.empty()) {
                cookie_str += "; ";
            }
            cookie_str += it.key() + "=" + it.value().get<std::string>();
        }
        request["cookies"] = cookie_str;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/r1/cpp_to_py/code_output/CookiesUtil.py
import json
import sys

class CookiesUtil:
    def __init__(self, cookies_file: str):
        self.cookies_file = cookies_file
        self.cookies = {}

    def get_cookies(self, response: dict):
        if "cookies" in response:
            # response["cookies"] is expected to be a dict mapping strings to strings
            self.cookies = response["cookies"]
        self._save_cookies()

    def load_cookies(self) -> dict:
        data = {}
        try:
            with open(self.cookies_file, 'r') as f:
                data = json.load(f)
        except FileNotFoundError:
            # file not present -> return empty dict (matches C++ behavior where file.open fails)
            pass
        except json.JSONDecodeError as e:
            print(f"Error reading JSON file: {e}", file=sys.stderr)
        except Exception as e:
            print(f"Error reading JSON file: {e}", file=sys.stderr)
        return data

    def _save_cookies(self) -> bool:
        try:
            with open(self.cookies_file, 'w') as f:
                json.dump(self.cookies, f, indent=4)
            return True
        except Exception as e:
            print(f"Error writing JSON file: {e}", file=sys.stderr)
            return False

    def set_cookies(self, request: dict):
        parts = []
        for key, value in self.cookies.items():
            parts.append(f"{key}={value}")
        request["cookies"] = "; ".join(parts)